search/query-features-with-arcade-expression/src/main/java/com/esri/samples/query_features_with_arcade_expression/QueryFeaturesWithArcadeExpressionSample.java

```diff
@@ -0,0 +1,198 @@
+              }
+            }});
+        } else if (map.getLoadStatus() == LoadStatus.FAILED_TO_LOAD) {
+          new Alert(Alert.AlertType.ERROR, "Map failed to load").show();
+        }
+      });
+
+      // add the map view to the stack pane
+      stackPane.getChildren().addAll(mapView);
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Evaluates an arcade expression at the point on the map that was clicked.
+   *
+   * @param point The point on the map which was clicked, in screen (rather than map) coordinates.
+   * @throws RuntimeException If unable to get results from futures due to interruption or concurrent execution.
+   */
+  private void evaluateArcadeExpression(Point2D point) throws RuntimeException {
+    // convert the passed point from screen coordinates to map coordinates, for use in displaying the callout
+    Point mapPoint = mapView.screenToLocation(point);
+
+    var identifyLayerResultFuture = mapView.identifyLayerAsync(policeBeatsLayer, point,
+      0, false);
+    identifyLayerResultFuture.addDoneListener(() -> {
+      if (identifyLayerResultFuture.isDone()) {
+        try {
+          var identifyLayerResult = identifyLayerResultFuture.get();
+
+          // only execute query if the police beats layer has data at the selected point
+          if (identifyLayerResult.getElements().size() != 0) {
+            ArcGISFeature feature = (ArcGISFeature) identifyLayerResult.getElements().get(0);
+
+            // show callout and text without data
+            mapView.getCallout().setTitle(calloutText);
+            mapView.getCallout().setDetail("");
+            mapView.getCallout().showCalloutAt(mapPoint);
+            mapView.getCallout().setVisible(true);
+
+            //setup arcade expression and evaluator
+            String expressionString = "var crimes = FeatureSetByName($map, 'Crime in the last 60 days');\n" +
+              "return Count(Intersects($feature, crimes));";
+            var arcadeExpression = new ArcadeExpression(expressionString);
+            var arcadeEvaluator = new ArcadeEvaluator(arcadeExpression, ArcadeProfile.FORM_CALCULATION);
+
+            // instantiate key/value pairs
+            var hashMap = new HashMap<String, Object>();
+            hashMap.put("$feature", feature);
+            hashMap.put("$map", mapView.getMap());
+
+            // evaluate the arcade expression asynchronously
+            var resultFuture = arcadeEvaluator.evaluateAsync(hashMap);
+
+            // wait for the expression to finish evaluating before attempting to access it
+            resultFuture.addDoneListener(() -> {
+              if (resultFuture.isDone()) {
+                try {
+                  // get result from async method and cast to int
+                  var arcadeEvaluationResult = resultFuture.get();
+                  var crimesCount = (int) ((double) arcadeEvaluationResult.getResult());
```

Comment:
to save this double cast, use Math.round when you are later calling the crime count.

or use it here

```suggestion
                  var crimesCount = Math.round((double) arcadeEvaluationResult.getResult());
```